WooCommerce/Classes/ViewRelated/Blaze/BlazeCampaignList/BlazeCampaignListViewModel.swift

```diff
@@ -149,6 +159,15 @@ private extension BlazeCampaignListViewModel {
             didShowIntroView = true
         }
     }
+
+    /// Clears application icon badge
+    ///
+    func resetNotificationBadgeCount() {
+        let kind: [Note.Kind] = [.blazeApprovedNote, .blazeRejectedNote, .blazeCancelledNote, .blazePerformedNote]
+        kind.forEach { kind in
+            pushNotesManager.resetBadgeCount(type: kind)
+        }
```

Comment:
❓ I think this means if you have multiple notifications for Blaze and you open one of them, the badge would be reset even though you haven't opened the others. Should it be a concern?

Reply:
Yes, I think this is OK. We do the same for orders. i.e. When the order screen appears, we clear the badge count irrespective of the number of new order notifications. [Code](https://github.com/woocommerce/woocommerce-ios/blob/trunk/WooCommerce/Classes/ViewRelated/Orders/OrdersRootViewController.swift#L129)